Set up Home page content including Landing

tyleryy opened the pull request:
## Overview

Created a rough draft of the apply button and layout for home page sections.

## Details

Home page layout uses Bootstrap Container and Rows to organize sections into a single column down the middle.

## Misc.
I want to make the apply button more "eye-catching" but for happy path it's good enough for now.

waalbert committed:
feat: apply button

* feat: create navigation bar

* feat: create footer

* refactor: font, dependencies, scss, footer

* fix: navbar on mobile

* refactor: added routes and other small changes

* feat: active page link has corresponding color

* feat: add ZotHacks logo

* fix: use responsive css and add bold weight

* chore: use absolute asset path for navbar background

* fix: logo moves upon navbar expansion and navbar is fixed to top of the screen

* refactor: clean up css and use bootstrap grid

* fix: remove horizontal scroll

* fix: switch Twitter to Hack and SVG to PNG

* fix: move Hack to front of footer

---------

Co-authored-by: Alexander Liu <a@alexanderliu.com>

feat: text centered

feat: added apply button svg to assets

Bootstrap Sass Organization (#17)

* feat: bootstrap sass organization

* fix: installed bootstrap

* fix: import font from Next instead of Google Fonts

* fix: update red hex value correctly

* linting

* fix: custom CSS property for font

* fix: sass partials

* fix: new sass path, incorporated navbar and footer

* fix: removed comment

---------

Co-authored-by: Tyler Yu <tyleryy@uci.edu>

feat: apply button

tyleryy committed:
Merge branch 'feature/home-page' of https://github.com/HackAtUCI/zothacks-site-2023 into feature/home-page

tyleryy committed:
fix: button dimensions and color

tyleryy committed:
feat: landing page sections

tyleryy committed:
fix: removed extraneous "as" prop

tyleryy committed:
Merge branch 'main' into feature/home-page

tyleryy committed:
fix: re-added button color

tyleryy committed:
feat: folder organization + Apply Button component

samderanova commented on apps/site/src/views/Landing/Landing.tsx:
```diff
@@ -1,9 +1,37 @@
 		<main>
-			<h1>ZotHacks 2023</h1>
+			<Container className="justify-content-center text-center">
+				<Row>
+					<h1>ZotHacks 2023</h1>
+				</Row>
```
No need to use a `Row` here for the `h1`. A good rule of thumb to follow is to use them only when you want your content to be in columns.

samderanova commented on apps/site/src/views/Landing/Landing.tsx:
```diff
@@ -1,9 +1,37 @@
+				<Row>
+					<h1>ZotHacks 2023</h1>
+				</Row>
+				<Row>
+					<ApplyButton />
+				</Row>
+				<Row>
+					<AboutSection />
+				</Row>
+				<Row>
+					<MentorSection />
+				</Row>
+				<Row>
+					<SponsorSection />
+				</Row>
+				<Row>
+					<FAQSection />
+				</Row>
```
For these, it might be better to use the `Row` component within the component's code instead of here, especially since in most cases, Bootstrap grid components will be used in them.

tyleryy replied:
I feel like it would better to keep the here because that way the components are more containerized and scoped. If we need to have more grid components, we can define a new container within each component. At least, I think this way is better. If not, let me know and I'll change it!

tyleryy replied:
I added the rows to prevent the components from being inline. I thought it was a good way to separate each component in a singular column. If there is a better way or this is not our desired behavior, pls let me know!


### With Row
<img width="1045" alt="Screen Shot 2023-09-15 at 5 23 40 PM" src="https://github.com/HackAtUCI/zothacks-site-2023/assets/66880934/5017bf9f-97cd-43f2-95f7-8321d20cc3b2">

### Without Row
<img width="511" alt="Screen Shot 2023-09-15 at 5 26 05 PM" src="https://github.com/HackAtUCI/zothacks-site-2023/assets/66880934/ad8dde4b-086b-487a-a527-99d874dcc821">

taesungh replied:
If the sections are actual elements and not just text nodes, they will stack vertically by default, no container or rows needed.

samderanova commented on apps/site/src/views/Landing/components/ApplyButton/ApplyButton.module.scss:
```diff
@@ -0,0 +1,33 @@
+.applyButton {
+	position: relative;
+	transform: skew(-30deg);
+	height: 80px;
+	width: 250px;
```
Nitpick: compared to the Figma, this parallelogram looks a bit too tall.

tyleryy replied:
I changed the text from a h2 to a h1 to make it fill the button up more and look closer to the Figma project. Let me know if it looks good!

samderanova commented on apps/site/src/views/Landing/components/ApplyButton/ApplyButton.tsx:
```diff
@@ -0,0 +1,16 @@
+import styles from "./ApplyButton.module.scss";
+
+import Button from "react-bootstrap/Button";
```
Nitpick: file imports should usually be after any imports from packages.

samderanova reviewed (changes requested):
Great start so far! Just a few suggestions:

tyleryy committed:
fix: changed apply button text to h1

tyleryy committed:
fix: removed row

taesungh commented on apps/site/src/assets/images/ApplyButton.svg:
Chore: asset is unused

taesungh commented on apps/site/src/views/Landing/components/ApplyButton/ApplyButton.tsx:
```diff
@@ -0,0 +1,15 @@
+				className={styles.applyButton}
+				href="https://hack.ics.uci.edu/" // TODO Change to application site link
+			>
+				<h1 className={styles.applyButtonText}>Apply!</h1>
```
Issue: non-semantic heading usage. Bootstrap provides utility classes to match heading styles, e.g. `.h2`

tyleryy replied:
Is this still an issue? I am a bit confused on what you would like me to change.

taesungh replied:
For proper document semantics, Heading elements should be used only for actual headings
> Do not use heading elements to resize text. ([Usage notes, Section Heading elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#usage_notes))

You could switch switch to something like a `span` element with the `.h2` class

taesungh commented on apps/site/src/views/Landing/components/ApplyButton/ApplyButton.module.scss:
```diff
@@ -0,0 +1,33 @@
+
+	&:hover {
+		border: black solid 6px;
+		background-color: utils.$yellow;
```
Thought: it may be better to define a proper button variant to handle the other states. Currently, the `:active` state reverts to Bootstrap's primary variant.

tyleryy replied:
How would I you like me to do this?

tyleryy replied:
I added a `warning` variant to the Button

tyleryy replied:
I just switched to regular button.

taesungh replied:
Using a `button` for a link is not appropriate. I'll handle this.

tyleryy replied:
Is it not good practice to put an `<a>` tag wrapped around a `<button>` ?

tyleryy replied:
I switched to a `div` instead. I didn't know that `a` tags shouldn't wrap interactive content per HTML W3C rules. So, thanks for the catch! Feel free to request or make any more changes, though, if the changes are not adequate!

taesungh commented on apps/site/src/views/Landing/components/ApplyButton/ApplyButton.module.scss:
```diff
@@ -0,0 +1,33 @@
+	margin: 0px;
+	border: none;
+	background-color: utils.$yellow;
+	transition: all 0.1s ease;
```
Suggestion: transitioning all properties may lead to unexpected behavior. It may be better to specify which properties should undergo transitions.

tyleryy replied:
One note: I used `text-decoration: underline` to underline on hover, but a transition cannot be applied to it.

taesungh reviewed (changes requested):
Decent start on the button design. A few organizational and usage notes.

taesungh commented on apps/site/src/views/Landing/components/ApplyButton/ApplyButton.module.scss:
```diff
@@ -0,0 +1,33 @@
+	width: 250px;
+	margin: 0px;
+	border: none;
+	background-color: utils.$yellow;
```
Issue: the highlighter yellow is slightly transparent (70% opacity) in the provided design

tyleryy commented on apps/site/src/views/Landing/Landing.tsx:
```diff
@@ -1,9 +1,35 @@
+import Row from "react-bootstrap/Row";
+
+import ApplyButton from "./components/ApplyButton/ApplyButton";
+import AboutSection from "./components/AboutSection/AboutSection";
+import MentorSection from "./components/MentorSection/MentorSection";
+import SponsorSection from "./components/SponsorSection/SponsorSection";
+import FAQSection from "./components/FAQSection/FAQSection";
```
kk changed let me know if this looks good!

tyleryy committed:
fix: "apply button" enhancements

fix: removed bootstrap imports

fix: removed unused asset

fix: removed apply button container div

fix: added warning variant

fix: 0.7 opacity on apply button

fix: native HTML button instead of Bootstrap

fix: opacity applied to background only + height change

tyleryy commented on apps/site/src/lib/styles/_zothacks-theme.scss:
```diff
@@ -1,5 +1,10 @@
 // Common variables for site theme consistency
 
+// import bootstrap scss
+@import "bootstrap/scss/functions";
+@import "bootstrap/scss/variables";
+@import "bootstrap/scss/mixins";
+
```
I believe this was added when I pulled from main. I'll remove it

tyleryy commented on apps/site/src/views/Landing/components/ApplyButton/ApplyButton.module.scss:
```diff
@@ -0,0 +1,33 @@
+	width: 250px;
+	margin: 0px;
+	border: none;
+	background-color: utils.$yellow;
```
How would you like me to make this change?

samderanova replied:
Remove the `zothacks-theme` import at the top of `_bootstrap_utils.scss`. Then you can import the utils as `bootstrap` and then the ZotHacks theme as `theme`, so now it's more apparent that the colors are related to our theme instead of utils.

tyleryy commented on apps/site/src/views/Landing/components/ApplyButton/ApplyButton.module.scss:
```diff
@@ -0,0 +1,33 @@
+
+.applyButtonText {
+	color: black;
+	position: absolute;
+	top: 50%;
+	left: 55%;
+	transform: skew(30deg) translate(-50%, -50%);
```
noticed this as well, but I am not sure how to prevent it

tyleryy replied:
nvm, I used `outline` instead of `border` to fix it.

tyleryy committed:
fix: switched to span with h1 class

tyleryy committed:
fix: adjusted button width

tyleryy committed:
fix: switched to span selector to apply underline

taesungh committed:
Reorganize Landing into Home

- To match common naming, rename primary page to Home
- Landing is now the first view of the Home page
- Wrap primary layout children in `<main>`

taesungh committed:
Use container layout for Home page sections

- Add `Container` elements for sections with responsive vertical padding

tyleryy committed:
fix: separated theme and utils scss files

tyleryy committed:
fix: button text doesn't shift

tyleryy committed:
fix: switched from button to div

taesungh committed:
Use bs-btn variant to simplify ApplyButton design

- Use `bootstrap.button-variant` mixin for Apply button styles
- Specify empty variant to React Bootstrap
- Remove unnecessary sizing and layout styles

tyleryy committed:
feat: added box shadow to apply button

samderanova commented on apps/site/src/views/Home/components/ApplyButton/ApplyButton.tsx:
```diff
@@ -0,0 +1,15 @@
+		<Button
+			className={styles.applyButton}
+			href="https://hack.ics.uci.edu/" // TODO: Change to application site link
+			variant=""
```
Nitpick: No need to set `variant` to an empty string

taesungh replied:
React Bootstrap will apply `.btn-primary` without it. I suppose the custom variant should override everything, but to be safe, better to not worry about the primary variant.

samderanova reviewed (changes requested):
Thanks for the hard work on this! As a suggestion, we could also make a custom `Container` component that has `as` set to `"section"` by default so we don't have to keep typing it out when we need to and have the custom padding.

Also, one small change to make:
